joomla/joomla-cms
# [5.1] Add Global Setting for Form Layout option to custom fields

Pull Request for Issue #37162. Follow-up to PR #36551 and issue #36548.

### Purpose of the PR

As discussed in #37162, this allows users to set a global preference for the HTML Select for custom fields of types: list, SQL, radio, integer.

I would have just removed the option, as the "raw" select is just ugly, but @laoneo noted that we need to keep it for Backwards Compatibility.

### Summary of Changes

- Added plugin settings for all relevant fields so that the field inherits from the global options by default (this allows users to set a global preference instead of having to set it for each field they create)
- The global option will still use the old "raw" style by default (to keep Backwards Compatibility)
- Implemented the same setting for the Integer field type as well (it was forgotten about in PR 36551 )
- Renamed the option to "Form Layout" for it to be more explicative
- Added new language strings in a shared language file to avoid duplicating them for all field types

### Testing Instructions

First create the custom field:
- Create Custom Field of type 'list' for the Content component. Make sure that the 'Multiple' setting is set to 'Yes'
- Go to Options->Form Options: Keep the 'Form Layout' as 'Use Settings from Plugin'

Then:
- Go to Plugin Manager, and in the plugin options for 'Fields - List', change the default 'Form Layout' option
- Edit an article and check the form display of the field, it will change based on your selection of 'Form Layout'

### Actual result BEFORE applying this Pull Request

There was no global setting to inherit from the plugin for 'Form Layout'

### Expected result AFTER applying this Pull Request

There is now a global setting for 'Form Layout' that controls the default value when not specified in an individual field.

Changing the global setting will change the form layout of the field.
